File: doctorClinic/doctor_clinic_crawler.py
from settings import crawler_setting as cs
import requests

class DoctorClinicCrawler:
    def __init__(self, default_url, default_headers):
        self.url = default_url
        self.headers = default_headers

    def requestDoctorClinic(self, region):
        payload = cs.getRequestForm(region)
        response = requests.request("POST", self.url, headers=self.headers, data=payload)
        return response.text

File: dart/lib/src/doctorClinic/doctor_clinic_crawler.dart
import 'dart:convert';
import 'dart:io';

import 'package:getClinicDoctor_dart/resource/crawler_setting.dart' as cs;
import 'package:http/http.dart' as http;
import 'package:cp949/cp949.dart' as cp949;


class DoctorClinicCrawler {
  final String _url;
  //String get url => _url;
  final Map<String, String> _headers;
  //Map<String, String> get headers => _headers;

  DoctorClinicCrawler(this._url, this._headers);

  Future<String> requestDoctorClinic(String region) async {
    var payload = cs.getRequestForm(region);
    var response = await http.post(
        Uri.parse(_url),
        headers: _headers,
        body: payload,
        encoding: Encoding.getByName('utf-8'));

    return cp949.decodeString(response.body);
  }
}
